cmd/kpod/login.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/containers/image/pkg/authentication"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+const defaultServer = "https://registry-1.docker.io"
+
+var (
+	loginFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "password, p",
+			Usage: "Password",
+		},
+		cli.StringFlag{
+			Name:  "username, u",
+			Usage: "Username",
+		},
+	}
+	loginDescription = "Login to a container registry on a specified server."
+	loginCommand     = cli.Command{
+		Name:        "login",
+		Usage:       "login to a container registry",
+		Description: loginDescription,
+		Flags:       loginFlags,
+		Action:      loginCmd,
+		ArgsUsage:   "SERVER",
+	}
+)
+
+// loginCmd uses the authentication package to store a user's authenticated credentials
+// in an auth.json file for future use
+func loginCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) > 1 {
+		return errors.Errorf("too many arguments, login takes only 1 argument")
+	}
+	var server string
+	if len(args) == 1 {
```

Comment:
User expectation is to omit 'https://' scheme from registry arg. I think it's safe to prepend 'https://' to any URL passed in unless they provide it.

Reply:
:+1:

Reply:
This is happening in the authentication package under `getFullURL`